### PR TITLE
Adds nginx test on container restart and makes use of EE_DOCKER methods

### PR DESCRIPTION
**Pull request by geekodour**
* removed `run_compose_command` and replaced the required with methods added in https://github.com/EasyEngine/easyengine/pull/1428
* also closes https://github.com/rtCamp/sys/issues/96 by adding nginx test before container restart

this PR is tightly related to the changes made in https://github.com/EasyEngine/easyengine/pull/1428

Signed-off-by: Hrishikesh Barman <hrishikeshbman@gmail.com>

### CODE REVIEW

File: src/helper/class-ee-site.php [hunk 1]
```diff
@@ -506,16 +506,23 @@ public function restart( $args, $assoc_args, $whitelisted_containers = [] ) {
...
 		} else {
 			$containers = array_keys( $assoc_args );
 		}
 
-		foreach ( $containers as $container ) {
-			\EE\Site\Utils\run_compose_command( 'restart', $container );
+		if ( in_array( 'nginx', $containers ) ) {
+			$nginx_test_command = "sh -c 'nginx -t'";
+			if ( ! \EE_DOCKER::docker_compose_exec( $this->site_data['site_fs_path'], 'nginx', $nginx_test_command ) ) {
+				throw new \Exception( 'There was some error in docker-compose exec.' );
```
Comment (mrrobot47) [inline src/helper/class-ee-site.php]: A better error message on failure would be: `Nginx configuration test failed. Please check the site's nginx config.`
`There was some error in docker-compose exec` is a very generic message and does not convey anything helpful to the end user using EasyEngine.

File: src/helper/class-ee-site.php [hunk 2]
```diff
@@ -625,36 +632,28 @@ public function reload( $args, $assoc_args, $whitelisted_containers = [], $reloa
...
 
 		$this->site_data = get_site_info( $args );
 
-		chdir( $this->site_data['site_fs_path'] );
-
 		if ( $all || $no_service_specified ) {
-			$this->reload_services( $whitelisted_containers, $reload_commands );
+			foreach ( $whitelisted_containers as $container ) {
+				if ( ! \EE_DOCKER::docker_compose_exec( $this->site_data['site_fs_path'], $container, $reload_commands[ $container ] ) ) {
+					throw new \Exception( 'There was some error in docker-compose exec.' );
```
Comment (mrrobot47) [inline src/helper/class-ee-site.php]: Again something useful can be added here like: `Reloading service for $container failed.`

File: src/helper/site-utils.php [hunk 3]
```diff
@@ -448,27 +449,13 @@ function restart_site_containers( $site_fs_path, $containers ) {
...
-function run_compose_command( $action, $container, $action_to_display = null, $service_to_display = null ) {
-
-	$display_action  = $action_to_display ? $action_to_display : $action;
-	$display_service = $service_to_display ? $service_to_display : $container;
-
-	EE::log( ucfirst( $display_action ) . 'ing ' . $display_service );
-	EE::exec( "docker-compose $action $container", true, true );
+	$stopped = \EE_DOCKER::docker_compose_stop( $site_fs_path, $all_containers );
+	$forcermd = \EE_DOCKER::docker_compose_forcerm( $site_fs_path, $all_containers );
+	if ( ! (stopped && forcermd) ) {
```
Comment (mrrobot47) [inline src/helper/site-utils.php]: Missing `$` in variable names. Also, fix phpcs issues.

**Comment by kirtangajjar** [comment]
Closing this PR as there are multiple issues and has been left unresolved > 1 year

**Pull request closed without merging**